src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java

```diff
@@ -0,0 +1,79 @@
+package com.osiris.autoplug.client.tasks.updater.search;
+
+import com.google.gson.JsonObject;
+import com.osiris.autoplug.client.tasks.updater.plugins.MinecraftPlugin;
+import com.osiris.autoplug.client.tasks.updater.search.SearchResult;
+import com.osiris.autoplug.client.utils.UtilsURL;
+import com.osiris.jlib.json.Json;
+import com.osiris.jlib.logger.AL;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class CustomUpdateCheck {
+
+    private boolean isInt(String s) {
+        try {
+            Integer.parseInt(s);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public SearchResult doCustomCheck(MinecraftPlugin plugin) {
+        SearchResult sr;
+        if (plugin.getCustomCheckURL != "FORCE") {
+            sr = checkUpdate(plugin)
+        } else {
+            sr = new SearchResult(null, 1, "0.0.0", null, ".jar", null, null, false);
+        };
+        return sr;
+    }
+
+    private SearchResult checkUpdate(MinecraftPlugin plugin) {
```

Comment:
Looks good, please also create a test to show this is working.
Check ./src/java/test for existing examples.

Reply:
I do not understand the examples and thus have no clue what to do for this

Reply:
@vengestone-dragoon Yeah forget that, those tests aren't really useful in this case.
Please create a new test that runs doCustomCheck() to see if it works correctly

Reply:
Ok, i think i have done that now